client/src/components/v2/CameraRecordingStatus.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { getInfoFromPayload, useCameraRecordingStatus, initCameraStatus } from 'api/v2/camera';
+
+/**
+ * @typedef {object} CameraRecordingStatusProps
+ * @property {'primary'|'secondary'} device Camera screen
+ */
+
+/**
+ *  Status of a camera
+ *
+ * @param {CameraRecordingStatusProps} props Props
+ * @returns {React.Component<CameraRecordingStatusProps>} Component
+ */
+export default function CameraRecordingStatus({ device }) {
+  const { status, lastPayload } = useCameraRecordingStatus(device);
```

Comment:
Since payload can be considered a protocol-specific unstable API, it's best not to depend on it from the display components. It's only used once here, so this can be refactored to expose the payload info rather than the payload itself

Reply:
I'm not sure what the problem is here. If there is no payload, then `getInfo/StatusFromPayload` will simply return `null`. Could you rephrase the issue?

Reply:
What I mean is that anything outside of the API shouldn't rely on the shape of payload. It shouldn't have access to payload at all, since the actual contents of the payload is subject to change. Instead, the display components only have access to processed output of the API, as that can remain stable. When you use `useCameraRecordingStatus` in the display component, you don't actually care what the payload is, just the status, so why return the payload? Everything regarding the payload can be wrapped up inside the API.

Reply:
`lastPayload` has been replaced with `payload` and is of type `CameraRecordingStatusPayload` which is the payload string parsed and formatted into an object.
4376700